src/librustc_metadata/creader.rs

```diff
@@ -550,9 +550,10 @@ impl<'a> CrateLoader<'a> {
         dep_kind: DepKind,
     ) -> CrateNumMap {
         debug!("resolving deps of external crate");
-        if crate_root.is_proc_macro_crate() {
-            return CrateNumMap::new();
-        }
+
+        // Note that we need to resolve deps for proc-macro crates (just like normal crates)
+        // since we may need to decode `Span`s that reference the `CrateNums`
+        // of transitive dependencies
```

Comment:
Nit: this comment looks unnecessary, like "look, nothing unusual is happening here, move along". It's the removed special case that required a comment, but didn't have it.

Reply:
We still special-case proc macros is some places, so I thought it was worthwhile to explain why we do this (in case someone ever thinks of re-adding this an an optimization).

Reply:
There's nothing here about proc macros anymore though? I would agree with @petrochenkov that this comment seems more confusing than helpful. We should have a test that ensures this doesn't get re-added?